src/systems/physics/Physics.cc

```diff
@@ -4165,6 +4179,66 @@ void PhysicsPrivate::UpdateCollisions(EntityComponentManager &_ecm)
       });
 }
 
+//////////////////////////////////////////////////
+void PhysicsPrivate::UpdateRayIntersections(EntityComponentManager &_ecm)
+{
+  GZ_PROFILE("PhysicsPrivate::UpdateRayIntersections");
+  // Quit early if the MultiRayIntersections component hasn't been created.
+  // This means there are no systems that need contact information
+  if (!_ecm.HasComponentType(components::MultiRayIntersections::typeId))
+    return;
+
+  // Assume that there is only one world entity
+  Entity worldEntity = _ecm.EntityByComponents(components::World());
+
+  if (!this->entityWorldMap.HasEntity(worldEntity))
+  {
+    gzwarn << "Failed to find world [" << worldEntity << "]." << std::endl;
+    return;
+  }
+
+  auto worldRayIntersectionFeature =
+      this->entityWorldMap.EntityCast<RayIntersectionFeatureList>(worldEntity);
```

Comment:
You should check if the cast was successful. If the feature isn't available in the physics engine, the pointer will be `nullptr`.

Reply:
Fixed in https://github.com/gazebosim/gz-sim/pull/2514/commits/34cbfd66e52c57028c8cc35cf19a917e6db41bb2